src/app/modules/main/wallet_section/activity/controller.nim

```diff
@@ -0,0 +1,131 @@
+import NimQml, logging, std/json, sequtils, sugar, options
+
+import model
+import entry
+
+import ../transactions/item
+import ../transactions/module as transactions_module
+
+import backend/activity as backend_activity
+import backend/backend as backend
+import backend/transactions
+
+import app_service/service/transaction/service as transaction_service
+
+proc toRef*[T](obj: T): ref T =
+  new(result)
+  result[] = obj
+
+QtObject:
+  type
+    Controller* = ref object of QObject
+      model: Model
+      transactionsModule: transactions_module.AccessInterface
+      currentActivityFilter: backend_activity.ActivityFilter
+
+  proc setup(self: Controller) =
+    self.QObject.setup
+
+  proc delete*(self: Controller) =
+    self.QObject.delete
+
+  proc newController*(transactionsModule: transactions_module.AccessInterface): Controller =
+    new(result, delete)
+    result.model = newModel()
+    result.transactionsModule = transactionsModule
+    result.currentActivityFilter = backend_activity.getIncludeAllActivityFilter()
+    result.setup()
+
+  proc getModel*(self: Controller): QVariant {.slot.} =
+    return newQVariant(self.model)
+
+  QtProperty[QVariant] model:
+    read = getModel
+
+  # TODO: move it to service, make it async and lazy load details for transactions
+  proc backendToPresentation(self: Controller, backendEnties: seq[backend_activity.ActivityEntry]): seq[entry.ActivityEntry] =
+    var multiTransactionsIds: seq[int] = @[]
+    var transactionIdentities: seq[backend.TransactionIdentity] = @[]
+    var pendingTransactionIdentities: seq[backend.TransactionIdentity] = @[]
+
+    # Extract metadata required to fetch details
+    # TODO: temporary here to show the working API. Will be done as required on a detail request from UI
+    for backendEntry in backendEnties:
+      case backendEntry.transactionType:
+        of MultiTransaction:
```

Comment:
It will probably take some refactoring, but at some point it would IMO be better to have `SimpleTransaction` and `PendingTransaction` simply be particular cases of the generic `MuliTransaction` struct (which I would give a different name btw :D)